.github/workflows/fetch_stats.yml

```diff
@@ -38,9 +68,35 @@ jobs:
       - name: execute py script 
         run: |
           echo "PYTHONPATH=src" >> $GITHUB_ENV
-          python src/nsofetch/fetch_all.py
-          
-      - name: commit files
+          python src/nsofetch/fetch_${{ matrix.country }}.py ${{ matrix.index }}
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.country }}-${{ matrix.index }}
+          path: data/${{ matrix.country }}_inflation_${{ matrix.index }}.csv
+
+  deploy:
+    if: ${{ always() }}
+    needs: [fetch]
```

Comment:
`needs: [fetch]` means this job is reliant on the previous one (i.e. it can’t start until all scrapers have completed).

`if: ${{ always() }}` means even if the previous job fails (e.g. one or more scraper errors), it’ll still run.